core/src/main/java/com/linecorp/armeria/common/multipart/DefaultMultipart.java

```diff
@@ -128,30 +132,36 @@ public void subscribe(Subscriber<? super HttpData> subscriber, EventExecutor exe
 
     @Override
     public HttpRequest toHttpRequest(RequestHeaders requestHeaders) {
-        requireNonNull(requestHeaders, "requestHeaders");
-        MediaType contentType = requestHeaders.contentType();
-        if (contentType != null) {
-            checkArgument("multipart".equals(contentType.type()),
-                          "Content-Type: %s (expected: multipart content type)", contentType);
-            contentType = contentType.withParameter("boundary", boundary());
-        } else {
-            contentType = MediaType.MULTIPART_FORM_DATA.withParameter("boundary", boundary());
-        }
-
-        final MediaType finalMediaType = contentType;
-        final RequestHeaders updated = requestHeaders.withMutations(builder -> {
-            builder.addObject(HttpHeaderNames.CONTENT_TYPE, finalMediaType);
-        });
-        return HttpRequest.of(updated, this);
+        final RequestHeaders requestHeadersWithBoundary = injectBoundary(boundary, requestHeaders);
+        return HttpRequest.of(requestHeadersWithBoundary, this);
     }
 
     @Override
     public HttpRequest toHttpRequest(String path) {
         requireNonNull(path, "path");
-        final MediaType contentType = MediaType.MULTIPART_FORM_DATA.withParameter("boundary", boundary());
-        return HttpRequest.of(
-                RequestHeaders.of(HttpMethod.POST, path,
-                                  HttpHeaderNames.CONTENT_TYPE, contentType.toString()), this);
+        final MediaType contentType = MediaType.MULTIPART_FORM_DATA.withParameter(BOUNDARY_PARAMETER,
+                                                                                  boundary());
+        final RequestHeaders requestHeaders = RequestHeaders.builder(HttpMethod.POST, path)
+                                                            .contentType(contentType)
+                                                            .build();
+        return HttpRequest.of(requestHeaders, this);
+    }
+
+    @Override
+    public HttpResponse toHttpResponse(ResponseHeaders responseHeaders) {
+        final ResponseHeaders responseHeadersWithBoundary = injectBoundary(boundary, responseHeaders);
+        return HttpResponse.of(responseHeadersWithBoundary, this);
+    }
+
+    @Override
+    public HttpResponse toHttpResponse(HttpStatus status) {
+        requireNonNull(status, "status");
+        final MediaType contentType = MediaType.MULTIPART_FORM_DATA.withParameter(BOUNDARY_PARAMETER,
+                                                                                  boundary());
```

Comment:
Could we call either `boundary` or `boundary()` consistently?

Reply:
Sure, will fix this